src/utils/index.ts

```diff
@@ -189,3 +190,10 @@ export const validateIdentityName = (identityName: string): IdentityNameValidity
 
   return null
 }
+
+export const checkIdentityNameAvailibility = async (name: string, subdomain = 'id.blockstack') => {
+  const url = `https://registrar.blockstack.org/v1/names/${name.toLowerCase()}.${subdomain}`
+  const resp = await fetch(url)
+  const data = await resp.json()
+  return data
```

Comment:
We should probably have try/catch here

Reply:
What would it return on the error case? Don't we want the consumer to handle the error?

Reply:
yeah, I suppose we would want the consumer to handle the error. I think it'd be good to test and see all the outcomes. idk if it makes sense, but maybe we can do something better here with TS, to type any of the errors, etc. make the DX better than what it in the registrar.

Reply:
I'll type the return value

Reply:
Yeah since we are not throwing for the other errors, we should probably not throw for this error, either. Which means we need a try/catch for the 404